lib/widgets/twake_components/twake_loading/flashing_circle_widget.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+
+class FlashingCircle extends StatelessWidget {
+  const FlashingCircle({
+    super.key,
+    required this.index,
+    required this.repeatingController,
+    required this.dotIntervals,
+    required this.flashingCircleBrightColor,
+    required this.flashingCircleDarkColor,
+  });
+
+  final int index;
+  final AnimationController repeatingController;
+  final List<Interval> dotIntervals;
+  final Color flashingCircleDarkColor;
+  final Color flashingCircleBrightColor;
+
+  @override
+  Widget build(BuildContext context) {
+    return AnimatedBuilder(
+      animation: repeatingController,
+      builder: (context, child) {
+        final circleFlashPercent = dotIntervals[index].transform(
+          repeatingController.value,
+        );
+        final circleColorPercent = sin(pi * circleFlashPercent);
+
+        return Container(
+          width: 8 * repeatingController.value,
+          height: 8 * repeatingController.value,

```

Comment:
what is 8?

Reply:
8 is the size of 3 "bubbles" in loading animation